Add import from gsheet for users

mrchrisadams opened the pull request:
This PR addresses #2 , to support the importing of a user's questionnaire responses into their profile. 

They have control over this, and it _appends_ info to avoid losing existing profile information - with the idea being that we don't want to lose any information someone has already added to their profile.

This doesn't include any attempt yet to fit this into the existing designs on figma - that's addressed with #4

mrchrisadams committed:
Add ability to import a single user via CLI

mrchrisadams committed:
Expose importer via the front end for staff / admins

EvanHahn commented on cl8/templates/pages/home.html:
```diff
@@ -35,6 +35,16 @@
               {% endif %}
             </div>
+            <div class="messages">
+              {% if messages %}
```
Should this conditional wrap the whole `<div>`?

mrchrisadams replied:
Yes, this is a good point. I'll make these changes later after work today before merging in.

EvanHahn commented on docs/setting-up-google-sheets.md:
```diff
@@ -0,0 +1,84 @@
+# Setting up integration with Google Sheets
```
Love this documentation.

EvanHahn commented on cl8/users/importers.py:
```diff
@@ -751,3 +754,49 @@ def add_cat_responses_to_profiles():
+    gsheet = gc.open_by_key(settings.GSPREAD_KEY)
+    responses_worksheet = gsheet.worksheet(CAT_RESPONSES_WORKSHEET)
+
+    matching_cell_for_email = responses_worksheet.find(email)
```
Should we handle the case where there are multiple rows with the same email? I could see that causing confusion, and maybe that should return an error.

mrchrisadams replied:
Yes, this is a very good point - we do have cases where we do indeed have more match. In this scenario, I think returning the last match would make sense as it would be the most recent submission.

EvanHahn commented on cl8/users/api/views.py:
```diff
@@ -287,6 +290,51 @@ def get(self, *args, **kwargs):
+                )
+                return redirect(profile.get_absolute_url())
+            except NoMatchingCAT:
+                # add message to say no matching CAT found, and redirect back
+                # to the profile view url
+                messages.add_message(
+                    request,
+                    messages.WARNING,
+                    (
+                        "Sorry, we couldn't find any matching joining form information"
+                        " for this user. "
+                    ),
+                )
+
+            return redirect(profile.get_absolute_url())
```
Could we combine this with the identical code below?

EvanHahn commented on cl8/users/importers.py:
```diff
@@ -751,3 +754,49 @@ def add_cat_responses_to_profiles():
+    matching_cell_for_email = responses_worksheet.find(email)
+
+    if not matching_cell_for_email:
+        raise NoMatchingCAT
```
Should this raise an instance of the error instead of the class itself? (I don't know Python well enough to know whether this works.)

mrchrisadams replied:
D'oh! yes. Something like this would make much more sense.

```python
NoMatchingCAT(email)
```

EvanHahn reviewed (approved):
LGTM other than some small comments!

mrchrisadams commented:
hi @EvanHahn thanks for the code review - I appreciate you catching a few things :D.

I'll merge this in tonight, as I'm almost done with the importer for the existing Airtable directory meaning we'd be able to import for both kinds of data that might relate to a given member in CAT.

mrchrisadams committed:
Move js to own file for profile

mrchrisadams committed:
Small changes to importer since review

1. Always fetch last matching email, in case we have multiple matches
2. Raise exception properly
3. Add horizontal separator after adding content

mrchrisadams committed:
Add changes as per Evan's code review
